electricitymap/capacity_parsers/EMBER.py

```diff
@@ -0,0 +1,176 @@
+    "ZA",
+]
+
+
+def map_variable_to_mode(row: pd.Series) -> pd.DataFrame:
+    zone = row["zone_key"]
+    variable = row["mode"]
+    if zone in SPECIFIC_MODE_MAPPING:
+        if variable in SPECIFIC_MODE_MAPPING[zone]:
+            row["mode"] = SPECIFIC_MODE_MAPPING[zone][variable]
+        else:
+            row["mode"] = EMBER_VARIABLE_TO_MODE[variable]
+    else:
+        row["mode"] = EMBER_VARIABLE_TO_MODE[variable]
+    return row
+
+
+def get_data_from_url(session: Session) -> pd.DataFrame:
+    base_url = "https://ember-climate.org"
+    yearly_catalogue_url = base_url + "/data-catalogue/yearly-electricity-data/"
+    r: Response = session.get(yearly_catalogue_url)
+    soup = BeautifulSoup(r.text, "html.parser")
+    csv_link = soup.find("a", {"download": "yearly_full_release_long_format.csv"})[
+        "href"
+    ]
+    r_csv: Response = session.get(base_url + csv_link)
+    df = pd.read_csv(io.StringIO(r_csv.text))
+    return df
+
+
+def format_ember_data(df: pd.DataFrame, year: int) -> pd.DataFrame:
+    df_filtered = df.loc[df["Area type"] == "Country"].copy()
+    df_filtered = df_filtered.loc[df_filtered["Year"] == year]
+    if df_filtered.empty:
+        raise ValueError(f"No data for year {year}")
+    df_filtered = df_filtered.loc[
+        (df_filtered["Category"] == "Capacity") & (df_filtered["Subcategory"] == "Fuel")
+    ]
+    # filter out Kosovo because it is not a country in pycountry
+    df_filtered = df_filtered.loc[df_filtered["Area"] != "Kosovo"]
+
+    df_filtered["country_code_iso2"] = df_filtered["Country code"].apply(
+        lambda x: pycountry.countries.get(alpha_3=x).alpha_2
+    )
+    df_filtered = df_filtered.loc[df_filtered["country_code_iso2"].isin(EMBER_ZONES)]
+
+    df_capacity = df_filtered[["country_code_iso2", "Year", "Variable", "Value"]]
+    df_capacity = df_capacity.rename(
+        columns={
+            "country_code_iso2": "zone_key",
+            "Year": "datetime",
+            "Variable": "mode",
+            "Value": "value",
+        }
+    )
+    df_capacity["datetime"] = df_capacity["datetime"].apply(lambda x: datetime(x, 1, 1))
+    df_capacity["value"] = df_capacity["value"] * 1000  # convert from GW to MW
+
+    df_capacity = df_capacity.apply(map_variable_to_mode, axis=1)
+
+    df_capacity = (
+        df_capacity.groupby(["zone_key", "datetime", "mode"])[["value"]]
+        .sum()
+        .reset_index()
+        .set_index(["zone_key"])
+    )
+    return df_capacity
+
+
+def get_capacity_dict_from_df(df_capacity: pd.DataFrame) -> dict:
+    all_capacity = {}
+    for zone in df_capacity.index.unique():
+        df_zone = df_capacity.loc[zone]
+        zone_capacity = {}
+        for i, data in df_zone.iterrows():
+            mode_capacity = {}
+            mode_capacity["datetime"] = data["datetime"].strftime("%Y-%m-%d")
+            mode_capacity["value"] = round(float(data["value"]), 0)
+            mode_capacity["source"] = "Ember, Yearly electricity data"
```

Comment:
As with other parsers, I think it's better to have this as a top level constant rather than hidden in the code.